### PR TITLE
docs: Documentation for Mava components

### LINKED ISSUE #648: [MAINTAIN] Component Documentation
### Please describe what needs to be maintained?
Mava components need documentation. Everything within `mava/components/jax`

### Describe the outcome you'd like
All classes / methods / functions should be clearly documented with explanations and descriptions of all args / returns

### PR DESCRIPTION
**Pull request by mmorris44**
## What?
Documented jax Mava components.
## Why?
Jax mava systems need documentation.
## How?
Filled out docstrings and added comments.
## Extra
Closes #648

### CODE REVIEW

File: mava/components/jax/training/losses.py [hunk 1]
```diff
@@ -58,7 +65,20 @@ def loss_grad_fn(
...
             behavior_values: Dict[str, jnp.ndarray],
         ) -> Tuple[Dict[str, jnp.ndarray], Dict[str, Dict[str, jnp.ndarray]]]:
-            """Surrogate loss using clipped probability ratios."""
+            """Surrogate loss using clipped probability ratios.
+
+            Args:
+                params: network parameters.
+                observations: agent observations.
+                actions: actions the agents took.
+                behaviour_log_probs: TODO(Matthew): not sure what this is.
```
Comment (siddarthsingh1) [inline mava/components/jax/training/losses.py:74]: @mmorris44 It's the log probability of the action taken at that timestep. `behavior_values` on line 77 is the estimated value from the critic at that timestep if I'm not mistaken? The name `behavior_values` might be a bit misleading tbh.

File: mava/components/jax/building/parameter_client.py [hunk 2]
```diff
@@ -50,19 +61,25 @@ def __init__(
...
+            config: ExecutorParameterClientConfig.
         """
 
         self.config = config
 
     def on_building_executor_parameter_client(self, builder: SystemBuilder) -> None:
-        """_summary_
+        """Create and store the executor parameter client.
+
+        Gets network parameters from store and registers them for tracking.
```
Comment (AsadJeewa) [inline mava/components/jax/building/parameter_client.py:75]: "registers them for tracking" is not so clear imo

File: mava/components/jax/building/reverb_components.py [hunk 3]
```diff
@@ -243,11 +246,17 @@ def on_building_data_server_start(self, builder: SystemBuilder) -> None:
         """Hook for adding reverb selector to builder store.
 
         This determines how experience will be experience from the replay table.
```
Comment (AsadJeewa) [inline mava/components/jax/building/reverb_components.py:248]: probably "will be sampled"

**Review by AsadJeewa — CHANGES_REQUESTED** [review]
Thanks for this @mmorris44 
Minor changes/ clarifications requested

File: mava/components/jax/training/losses.py (hunk 1 above)
Comment (mmorris44) [reply mava/components/jax/training/losses.py:74]: Thanks for the clarifications! I've added the info.

File: mava/components/jax/building/adders.py [hunk 4]
```diff
@@ -25,17 +25,22 @@
...
+            builder: SystemBuilder.
 
         Returns:
-            _description_
+            None.
         """
+
+    @staticmethod
+    def name() -> str:
+        """Static method that returns component name."""
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply mava/components/jax/building/adders.py:43]: Done, thanks! Was mostly putting this off due to laziness, but it was an easy fix.

File: mava/components/jax/building/adders.py [hunk 5]
```diff
@@ -25,17 +25,22 @@
 
 
 class Adder(Component):
+    """Abstract Adder component defining which hooks should be used."""
+
     @abc.abstractmethod
     def on_building_executor_adder(self, builder: SystemBuilder) -> None:
-        """[summary]"""
+        """Create the executor adder.
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply mava/components/jax/building/adders.py:32]: I'd rather leave that for another PR. Also, it's an inconsistency that, in my opinion, doesn't make a big difference at all

File: mava/components/jax/building/datasets.py [hunk 6]
```diff
@@ -33,29 +33,28 @@ def __init__(
         self,
         config: Any,
     ):
-        """_summary_
+        """Component creates a reverb table for the trainer.
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply mava/components/jax/building/datasets.py:36]: I don't see where it uses existing reverb tables though? It seems to create them from scratch

File: mava/components/jax/building/parameter_client.py (hunk 2 above)
Comment (mmorris44) [reply mava/components/jax/building/parameter_client.py:75]: Changed to "for updating"

File: mava/components/jax/building/datasets.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (AsadJeewa) [reply mava/components/jax/building/datasets.py:36]: so as far as I understand, the executor and trainer are both clients of the reverb table. The data server creates the tables. The executor adds data to the tables and the trainer samples from the tables via the dataset

File: mava/components/jax/building/parameter_client.py (hunk 2 above)
Comment (AsadJeewa) [reply mava/components/jax/building/parameter_client.py]: Thanks

File: mava/components/jax/building/adders.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (AsadJeewa) [reply mava/components/jax/building/adders.py:32]: Okay, that's fair. It doesn't make a difference beyond consistency

File: mava/components/jax/building/datasets.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (mmorris44) [reply mava/components/jax/building/datasets.py:36]: Ahh I see, thanks for the clarification. I've fixed the docstrings.

**Review by AsadJeewa — APPROVED** [review]
All good on my side! Thanks @mmorris44

**Review by siddarthsingh1 — APPROVED** [review]
@mmorris44 Looks good on my side 👍.

**Pull request merged**